src/views/bootablevolumes/actions/components/EditBootableVolumesModal.tsx

```diff
@@ -17,73 +19,96 @@ import TabModal from '@kubevirt-utils/components/TabModal/TabModal';
+import { readableSizeUnit } from '@kubevirt-utils/utils/units';
 import { K8sResourceCommon } from '@openshift-console/dynamic-plugin-sdk';
-import { Form, FormGroup, Grid, GridItem, PopoverPosition, TextArea } from '@patternfly/react-core';
+import {
+  Form,
+  FormGroup,
+  Grid,
+  GridItem,
+  PopoverPosition,
+  Select,
+  SelectOption,
+  SelectVariant,
+  TextArea,
+} from '@patternfly/react-core';
 
 import { BootableVolumeMetadata } from '../../utils/types';
-import { changeBootableVolumeMetadata, getInstanceTypesToSizesMap } from '../../utils/utils';
+import { changeBootableVolumeMetadata } from '../../utils/utils';
 
 type EditBootableVolumesModalProps = {
   dataSource: V1beta1DataSource;
   isOpen: boolean;
   onClose: () => void;
   preferences: V1alpha2VirtualMachineClusterPreference[];
-  instanceTypesNames: string[];
 };
 
 const EditBootableVolumesModal: FC<EditBootableVolumesModalProps> = ({
   dataSource,
   isOpen,
   onClose,
   preferences,
-  instanceTypesNames,
 }) => {
   const { t } = useKubevirtTranslation();
+
   const preferencesNames = useMemo(
     () => Object.keys(convertResourceArrayToMap(preferences)).sort((a, b) => a.localeCompare(b)),
     [preferences],
   );
 
-  const instanceTypesToSizesMap = useMemo(
-    () => getInstanceTypesToSizesMap(instanceTypesNames),
-    [instanceTypesNames],
-  );
-
   const initialParams = useMemo(() => {
     const instanceTypeLabel =
       dataSource?.metadata?.labels?.[DEFAULT_INSTANCETYPE_LABEL]?.split('.');
+    const initialCategory = Object.entries(categoryDetailsMap).find(
+      (category) => category[1].prefix === instanceTypeLabel?.[0],
+    );
 
     return {
       preference: dataSource?.metadata?.labels?.[DEFAULT_PREFERENCE_LABEL],
-      instanceType: instanceTypeLabel?.[0],
-      size: instanceTypeLabel?.[1],
+      instanceType: initialCategory,
+      size: initialCategory && instanceTypeLabel?.[1],
       description: dataSource?.metadata?.annotations?.[ANNOTATIONS.description],
     };
   }, [dataSource]);
 
   const [preference, setPreference] = useState<string>(initialParams.preference);
-  const [instanceType, setInstanceType] = useState<string>(initialParams.instanceType);
+  const [isInstanceTypeOpen, setIsInstanceTypeOpen] = useState(false);
+  const [instanceType, setInstanceType] = useState<string>(initialParams.instanceType?.[0]);
+  const [isSizeOpen, setIsSizeOpen] = useState(false);
   const [size, setSize] = useState<string>(initialParams.size);
   const [description, setDescription] = useState<string>(initialParams.description);
 
-  const sizeOptions = useMemo(
-    () => instanceTypesToSizesMap[instanceType] || [],
-    [instanceType, instanceTypesToSizesMap],
+  // update options for 'Size' dropdown according to chosen 'instanceType' in 'Default InstanceType' dropdown
+  const { instanceTypes }: CategoryDetails = useMemo(
+    () => (instanceType ? categoryDetailsMap[instanceType] : {}),
```

Comment:
This is the problem of naming the data we are working with, as I've already mentioned in another comment... I'm not gonna change that till we get to some convention for names of the data we are working with, regarding InstanceTypes.

Reply:
I understand that you don't like the naming as it is right now. That's still no excuse to use comments of self-documenting code. The naming can be updated if/when the nomenclature is clarified.

Reply:
Seems like we have different opinion on if my comment in this place is appropriate or not. I was thinking about these names and stuff for a long time, when working on this PR. And I decided to better put the comment in this place. I would go against myself and other good habits of one developer if I simply ignored the fact of our naming issue and not added that comment at all. Or if I used long ugly unreadable name like `DefaultInstanceTypeDropdownOptions`, because that would be the most appropriate to describe what is going on there 😇 With such long names everywhere in our code, we wouldn't help us much. I'll better add one readable comment, in this specific case. Do we have to make it ourselves even harder if we don't have to?

Once we get to some clear naming convention for InstanceTypes related data, the comment will become unnecessary and will be removed, together with updating related namings we use in our code.

We can change everything we want. That means we can add or remove any comment we want, too. That's nothing permanent in our code. So please take it as a temporary improvement of what we have just now. Btw, I'd like to see more comments in our code than we have now. Because there are the same naming issues also in other areas, and absence of comments in some places makes (not just) my life just harder, slows down the development... Using comments isn't bad. It's a tool. We should use the tools we have if they help us, if the actual upside of the used tool is higher than its downside. When using too many unnecessary comments - that's not helpful, of course. Unfortunately, this is obviously subjective when some comment is good to use and when not. I think there is no any clear way how to solve that universally, except hiring a different dev who would be more aligned with opinions and experience of other devs in the team 😄

Reply:
We should avoid long names if we can use a shorter one that communicates the necessary information, but using a longer name that communicates that information is superior to a comment as it will clear anywhere it's used in the code and not just in that one spot. In general, if a comment is just stating what the code is doing and the code isn't something that's unusually hard to parse (like some type of non-idiomatic code or regex), the solution is to refactor, not comment.

I just looked at it again and realized this block of code could be clearer if you renamed the `instanceType` state to `category` as suggested above.

```
  const instanceTypes: InstanceTypeSizeDetails[] = useMemo(
    () => (category ? categoryDetailsMap[category].instanceTypes : {}),
    [category],
  );
```

> I'll better add one readable comment, in this specific case. Do we have to make it ourselves even harder if we don't have to?

I see you use this reasoning a lot. It's better to avoid using a comment, if possible, than use it as a crutch to avoid using practices that aid code readability like good naming, extracting logic into methods that can benefit from good naming, etc.

> We can change everything we want. That means we can add or remove any comment we want, too. That's nothing permanent in our code. So please take it as a temporary improvement of what we have just now.

It's true that we can modify the code as we see fit. We should still always opt to make the best changes possible. In this case, there were better choices than the comment as I pointed out above.

> I'd like to see more comments in our code than we have now. Because there are the same naming issues also in other areas

If there are areas that you think need comments or that suffer from naming issues, then that means the code likely needs to be refactored. It's more likely that the issue is a code smell than a scenario where a comment is the best solution. I suggest you put together a PR with the comments you feel are necessary and let the team review it. It will likely result in lots of opportunity for refactoring.